services/compute/virtualmachine/virtualmachine.go

```diff
@@ -4,17 +4,19 @@
 package virtualmachine
 
 import (
+	"github.com/microsoft/moc-sdk-for-go/services/compute"
+	"github.com/microsoft/moc/pkg/auth"
 	"github.com/microsoft/moc/pkg/convert"
 	"github.com/microsoft/moc/pkg/errors"
-
-	"github.com/microsoft/moc-sdk-for-go/services/compute"
+	"github.com/microsoft/moc/pkg/validations"
 
 	"github.com/microsoft/moc/pkg/status"
 	wssdcloudcompute "github.com/microsoft/moc/rpc/cloudagent/compute"
+	wssdcloudproto "github.com/microsoft/moc/rpc/common"
 	wssdcommon "github.com/microsoft/moc/rpc/common"
 )
 
-func (c *client) getWssdVirtualMachine(vm *compute.VirtualMachine, group string) (*wssdcloudcompute.VirtualMachine, error) {
+func (c *client) getWssdVirtualMachine(vm *compute.VirtualMachine, group string, opType wssdcloudproto.Operation) (*wssdcloudcompute.VirtualMachine, error) {
```

Comment:
What operation are we restricting this from? Why are we adding the check for POST operation

Reply:
Below validations should be done only for post operation

Reply:
What are the implications if we pass the proxy value for other operations? I thought the other requests should be noop. Correct me if I am wrong

Reply:
reason why I added post operation check is because for other operations we get redacted values which fails the validations